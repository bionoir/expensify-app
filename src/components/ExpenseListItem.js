import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = ( { id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}>{description}</Link>
        <p>amount: {amount}, created at: {moment(createdAt).format('DD-MM-YYYY')}</p>
    </div>
);


export default ExpenseListItem;