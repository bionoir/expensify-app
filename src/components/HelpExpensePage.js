import React from 'react';

const HelpExpensePage = () => (
    <div>
        This is my Help expense page
    </div>
);

export default HelpExpensePage;