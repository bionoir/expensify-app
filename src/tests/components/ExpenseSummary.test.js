import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary correctly', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={1} expensesTotal={100}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with empty message', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={23} expensesTotal={23321322} />);
    expect(wrapper).toMatchSnapshot();
});