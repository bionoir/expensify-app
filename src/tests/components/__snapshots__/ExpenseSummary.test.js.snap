// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        CHF1.00
      </span>
       
    </h1>
    <div
      className="page-header__actions"
    >
      <[object Object]
        className="button"
        to="/create"
      >
        Add expense
      </[object Object]>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary with empty message 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        23
      </span>
       
      expenses
       totalling 
      <span>
        CHF233'213.22
      </span>
       
    </h1>
    <div
      className="page-header__actions"
    >
      <[object Object]
        className="button"
        to="/create"
      >
        Add expense
      </[object Object]>
    </div>
  </div>
</div>
`;
